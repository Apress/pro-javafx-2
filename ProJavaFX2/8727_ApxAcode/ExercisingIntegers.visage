/*
 * @test
 * @run
 */

var i = 1024;
var j = 2048;
var k = 15625;

println("i = {i}");
println("j = {j}");
println("k = {k}");

println("i + j = {i + j}");
println("i - j = {i - j}");
println("i * k = {i * k}");
println("k / i = {k / i}");
println("k mod j = {k mod j}");

var max = Integer.MAX_VALUE;
var min = Integer.MIN_VALUE;

println("max = {max}");
println("min = {min}");

println("max + 1 will overflow to {max + 1}");
println("min - 1 will overflow to {min - 1}");
println("max * min will overflow to {max * min}");

println("i = {i}, ++i = {++i}");
println("j = {j}, --j = {--j}");
println("k = {k}, k++ = {k++}");
println("k = {k}, k-- = {k--}");
