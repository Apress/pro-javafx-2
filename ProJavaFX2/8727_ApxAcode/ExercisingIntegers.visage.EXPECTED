i = 1024
j = 2048
k = 15625
i + j = 3072
i - j = -1024
i * k = 16000000
k / i = 15
k mod j = 1289
max = 2147483647
min = -2147483648
max + 1 will overflow to -2147483648
min - 1 will overflow to 2147483647
max * min will overflow to -2147483648
i = 1024, ++i = 1025
j = 2048, --j = 2047
k = 15625, k++ = 15625
k = 15626, k-- = 15626
